From 76c6800f47dabff89e6aee32c90d6064324868a9 Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston <jeremyhu@apple.com>
Date: Tue, 11 Oct 2011 00:33:35 -0700
Subject: [PATCH] setup: Do not strip trailing / from paths

real_path will strip the trailing / from provided paths.  This fixes
a regression introduced in 18e051a3981f38db08521bb61ccf7e4571335353

Signed-off-by: Jeremy Huddleston <jeremyhu@apple.com>
---

Patch sent upstream via email on 10/11/2011 to
carenas@sajinet.com.pe gitster@pobox.com

<rdar://problem/10238070>

 setup.c |   10 +++++++++-
 1 files changed, 9 insertions(+), 1 deletions(-)

diff --git a/setup.c b/setup.c
index 61c22e6..e3a8ae3 100644
--- a/setup.c
+++ b/setup.c
@@ -10,8 +10,16 @@ char *prefix_path(const char *prefix, int len, const char *path)
 	char *sanitized;
 	if (is_absolute_path(orig)) {
 		const char *temp = real_path(path);
-		sanitized = xmalloc(len + strlen(temp) + 1);
+		sanitized = xmalloc(len + strlen(temp) + 2);
 		strcpy(sanitized, temp);
+
+		temp = strrchr(path, '\0');
+		temp--;
+		if (*temp == '/') {
+			char *s = strrchr(sanitized, '\0');
+			s[0] = '/';
+			s[1] = '\0';
+		}
 	} else {
 		sanitized = xmalloc(len + strlen(path) + 1);
 		if (len)
-- 
1.7.6.1
